### PR TITLE
Issue # 520: add a "copy to clipboard" button near to locale translation

### PR DESCRIPTION
**Pull request by pascalchevrel**
* add clipboard.js as a dependency through composer
* use unicode scissors character for copy
* add styles for feedback message inserting a 'Translation copied!" message
* works in 2 or 3 locales search views
* js included in the base template as I think it may be useful in other views in the future

### CODE REVIEW

**Comment by pascalchevrel** [comment]
You can check the result on my VPS http://transvision.pascalc.net/?recherche=hous&repo=central&sourcelocale=fr&locale=it&search_type=strings

File: app/views/templates/base.php [hunk 1]
```diff
@@ -125,6 +125,20 @@
...
+     clipboard.on('success', function(e) {
+            e.trigger.setAttribute('data-title', 'Translation copied!');
+            e.trigger.setAttribute('class', 'clipboard tooltip');
+            setTimeout(function(arg1) {
+                arg1.trigger.setAttribute('class', 'clipboard')
+            }, 500, e);
+            e.clearSelection();
+        }
+    );
+ </script>
```
Comment (flodolo) [inline app/views/templates/base.php]: Can we move this to ```assets/base.js```? The idea would be to not have JavaScript code inside .php files, unless unavoidable (like at the end of the template, where PHP is needed to create the code dynamically).

File: app/classes/Transvision/ShowResults.php [hunk 2]
```diff
@@ -367,6 +371,7 @@ public static function resultsTable($search_results, $recherche, $locale1, $loca
                       <a class='bug_link' target='_blank' href='{$bz_link[1]}'>
                         &lt;report a bug&gt;
                       </a>
+                    <span class='clipboard' data-clipboard-target='#{$clipboard_target_string2}' alt='Copy to clipboard'>&#9986;</span>
```
Comment (flodolo) [inline app/classes/Transvision/ShowResults.php]: I confess I find the scissors icon a bit confusing, but there's not much of an alternative, unless we use an image with a clearer copy icon (like we're doing to [invert locales](https://transvision.mozfr.org/img/swap_icon_horiz_18x18.png))

File: app/views/templates/base.php [hunk 3]
```diff
@@ -125,6 +125,20 @@
...
   <script src="/assets/jquery/jquery.min.js?v=<?= VERSION ?>"></script>
+  <script src="/assets/clipboard.js/clipboard.js-built.js?v=<?= VERSION ?>"></script>
+  <script>
+     var clipboard = new Clipboard('.clipboard');
+     clipboard.on('success', function(e) {
+            e.trigger.setAttribute('data-title', 'Translation copied!');
+            e.trigger.setAttribute('class', 'clipboard tooltip');
+            setTimeout(function(arg1) {
+                arg1.trigger.setAttribute('class', 'clipboard')
+            }, 500, e);
```
Comment (flodolo) [inline app/views/templates/base.php]: I would display it a little longer (maybe 1s?)

File: web/style/transvision.css [hunk 4]
```diff
@@ -946,3 +946,43 @@ fieldset {
         font-size: 10px;
     }
 }
+
+/* Tooltips for copy-pasting of strings */
+.clipboard {
+    float: right;
+    font-size: 200%;
+    color: #000;
+}
```
Comment (flodolo) [inline web/style/transvision.css]: I would add `cursor: pointer;` to make clear it's clickable.

**Comment by flodolo** [comment]
It needs a bit of padding/margin at the smaller view port

schermata 2015-11-13 alle 09 56 18

Besides that it looks good to me with the few comments addressed.

File: web/style/transvision.css [hunk 5]
```diff
@@ -946,3 +946,52 @@ fieldset {
...
+.clipboard {
+    float: right;
+    font-size: 200%;
+    color: #000;
+    cursor: pointer;
+}
+
+@media only screen and (max-width: 460px) {
+    .clipboard img {
+        position:relative;
```
Comment (flodolo) [inline web/style/transvision.css]: nit: space after :

File: web/style/transvision.css [hunk 6]
```diff
@@ -946,3 +946,52 @@ fieldset {
...
+
+/* Tooltips for copy-pasting of strings */
+.clipboard {
+    float: right;
+    font-size: 200%;
+    color: #000;
+    cursor: pointer;
+}
+
+@media only screen and (max-width: 460px) {
```
Comment (flodolo) [inline web/style/transvision.css]: We should do this starting from 850px since we switch to the vertical layout there.

**Comment by zenorocha** [comment]
lol

**Pull request merged**